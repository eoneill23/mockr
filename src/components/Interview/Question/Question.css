.shadow {box-shadow: 0 16px 24px 2px rgba(0,0,0,0.15), 0 6px 30px 5px rgba(0,0,0,0.12), 0 8px 10px -7px rgba(0,0,0,0.2);}

.box-fix {
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

.card {
  background: #48a5a0;
  color: white;
  padding: 2rem;

  position: absolute;
  top: 2rem; bottom: 2rem;
  border-radius: 0.5rem;
  transition: transform 0.3s linear, opacity 0.3s linear;

  left: 5%; right: 5%; /* screen < 800px */
}

.card-question {
  font-size: 3rem;
  font-weight: 300;
  width: 90%;
  max-height: 35%;
  margin: 0;
  margin-top: 2rem;
}

.card-position {
  font-size: 4rem;
  font-weight: 400;
  margin: 0;
  position: absolute;
  top: 3rem;
  right: 3rem;
}

.header-notes {
  color: white;
  font-size: 2rem;
  font-weight: 300;
  position: absolute;
  bottom: calc(30% + 6rem); left: 2rem;
}

.card-response {
  resize: none;
  font-size: 1.5rem;
  color: white;
  background: transparent;
  border-top: 0.2rem solid white;
  border-bottom: 0.2rem solid white;
  border-left: none; border-right: none;

  position: absolute;
  bottom: 2rem; left: 2rem;
  width: calc(100% - 20rem);
  height: 40%;
}
.card-response:hover {background: rgba(255,255,255,0.1);}
.card-response:focus {outline: none; background: rgba(50,50,50,0.2);}

.card-score {
  font-size: 1.2rem;

  position: absolute;
  bottom: 10rem; right: 1.5rem;
  height: calc(40% - 8rem);
}
.card-score input {
  background: transparent;
}

.card-next-btn {
  position: absolute;
  left: calc(100% - 8rem);
  bottom: 3rem;
  width: 5rem; height: 5rem;
  background: #f03750;
  border: 0;
  border-radius: 5rem;
  color: #9b0528;
  font-size: 4.5rem;
  line-height: 2rem;
  margin: 0; padding: 0;
  cursor: pointer;

  display: none;
}
.card-next-btn:focus {outline: 0;}

@media only screen and (min-width: 800px) {
  .card {left: 20%; right: 20%;}
  .card-response {width: calc(100% - 13rem);}
  .card-next-btn {display: block;}
}
@media only screen and (min-width: 1400px) {
  .card {left: 30%; right: 30%;}
}




.card-focused {
  z-index: 0;
}

.card-first {
  z-index: -1;
  transform: translate(1rem, 0.5rem);
}

.card-second {
  z-index: -2;
  transform: translate(2rem, 1rem);
}

.card-third {
  z-index: -3;
  transform: translate(3rem, 1.5rem);
}

.card-hidden {
  z-index: -4;
  opacity: 0;
  transform: translate(4rem, 2rem);
}

.card-finished {
  z-index: 1;
  transform: translate(-100vw, 0);
}

