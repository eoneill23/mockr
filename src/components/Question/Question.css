.shadow {box-shadow: 0 16px 24px 2px rgba(0,0,0,0.15), 0 6px 30px 5px rgba(0,0,0,0.12), 0 8px 10px -7px rgba(0,0,0,0.2);}

.card {
  background: #48a5a0;
  color: white;
  padding: 2rem;

  position: fixed;
  top: 6rem; bottom: 2rem;
  left: 20%; right: 20%;
  border-radius: 0.5rem;
  transition: transform 0.3s linear, opacity 0.3s linear;
}

.card-question {
  font-size: 3rem;
  font-weight: 300;
  width: 90%;
  max-height: 40%;
}

.card-position {
  font-size: 4rem;
  font-weight: 400;
  margin: 0;
  position: absolute;
  top: 3rem;
  right: 3rem;
}



.card-focused {
  z-index: 0;
}

.card-first {
  z-index: -1;
  transform: translate(1rem, 0.5rem);
}

.card-second {
  z-index: -2;
  transform: translate(2rem, 1rem);
}

.card-third {
  z-index: -3;
  transform: translate(3rem, 1.5rem);
}

.card-hidden {
  z-index: -4;
  opacity: 0;
  transform: translate(4rem, 2rem);
}

.card-finished {
  z-index: 1;
  transform: translate(-100vw, 0);
}

