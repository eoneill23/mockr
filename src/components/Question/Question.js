import React from 'react';

export const Question = ({ body, notes, id, showDetails, detailed }) => {

  const isDetailed = () => {if (detailed) {return ' shown'} else {return ''}}
  return (
    <section className='question-card' onClick={e => showDetails(id)}>
      <h3>{body}</h3>
      <div className={'details' + isDetailed()}>
        <p>{notes}</p>
      </div>
    </section>
  )
}

export default Question;
