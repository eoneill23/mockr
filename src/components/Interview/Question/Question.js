import React from 'react';
import './Question.css';

export const Question = props => {
  const whereAmI = cur => {
    let pos = parseInt(props.pos);
    if (pos < cur) {
      return 'card-finished';
    } else if (pos === cur) {
      return 'card-focused';
    } else {
      if (pos === (cur + 1)) {
        return 'card-first';
      } else if (pos === (cur + 2)) {
        return 'card-second';
      } else if (pos === (cur + 3)) {
        return 'card-third';
      } else {
        return 'card-hidden';
      }
    }
  }

  return (
    <div className={'card shadow ' + whereAmI(props.cur)}>
      <h1 className='card-question'>Q: {props.question}</h1>
      <h1 className='card-position'>{props.pos}</h1>

      <form id={'card-response-' + props.id}>
        <h3 className='header-notes'>Notes:</h3>
        <textarea name='response' form={'card-response-' + props.pos} className='box-fix card-response' onChange={event => props.fs.note(props.id, event.target.value)}></textarea>
      </form>
      <button className='card-next-btn shadow' onClick={props.fs.next} value='ðŸ š'/>
    </div>
  );
}

export default Question;
