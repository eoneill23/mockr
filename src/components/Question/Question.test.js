import React from 'react';
import { render } from '@testing-library/react';
import Question from './Question';
import { MockedProvider } from '@apollo/react-testing';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Question', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(
      <MockedProvider addTypeName={false}>
        <Question cur={1} id={1} pos={2} question={'Hello'} scored={false} fs={{note: jest.fn(), score: jest.fn(), next: jest.fn(), skip: jest.fn()}}/>
      </MockedProvider>
    );
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});