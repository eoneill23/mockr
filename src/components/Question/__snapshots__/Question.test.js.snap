// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match the snapshot with all data passed in correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card shadow card-first"
      >
        <h1
          class="card-question"
        >
          Q: 
          Hello
        </h1>
        <h1
          class="card-position"
        >
          2
        </h1>
        <h3
          class="header-notes"
        >
          Notes:
        </h3>
        <textarea
          class="box-fix card-response"
          form="card-response-2"
          name="response"
        />
        <form
          class="card-score"
          id="score-form-1"
        >
          <input
            name="score"
            type="radio"
            value="1"
          />
          Unsatisfactory
          <br />
          <input
            name="score"
            type="radio"
            value="2"
          />
          Needs Work
          <br />
          <input
            name="score"
            type="radio"
            value="3"
          />
          Good
          <br />
          <input
            name="score"
            type="radio"
            value="4"
          />
          Exceptional
          <br />
        </form>
        <button
          class="card-next-btn shadow"
        >
          ðŸ š
        </button>
        <button
          class="card-skip-btn"
        >
          â†·
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card shadow card-first"
    >
      <h1
        class="card-question"
      >
        Q: 
        Hello
      </h1>
      <h1
        class="card-position"
      >
        2
      </h1>
      <h3
        class="header-notes"
      >
        Notes:
      </h3>
      <textarea
        class="box-fix card-response"
        form="card-response-2"
        name="response"
      />
      <form
        class="card-score"
        id="score-form-1"
      >
        <input
          name="score"
          type="radio"
          value="1"
        />
        Unsatisfactory
        <br />
        <input
          name="score"
          type="radio"
          value="2"
        />
        Needs Work
        <br />
        <input
          name="score"
          type="radio"
          value="3"
        />
        Good
        <br />
        <input
          name="score"
          type="radio"
          value="4"
        />
        Exceptional
        <br />
      </form>
      <button
        class="card-next-btn shadow"
      >
        ðŸ š
      </button>
      <button
        class="card-skip-btn"
      >
        â†·
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
